clang/include/clang/Sema/Sema.h

```diff
@@ -1759,6 +1759,33 @@ class Sema final {
     }
   };
 
+  /// Bitmask to contain the list of reasons a single diagnostic should be
+  /// emitted, based on its language.  This permits multiple offload systems
+  /// to coexist in the same translation unit.
+  enum DeviceDiagnosticReason {
```

Comment:
I would suggest renaming it to `DeviceDiagnosticContext`, but it is up to FE people to decide.

Reply:
I see. For creating a deferred diagnostic, 'reason' seems like an acceptable naming. However, in `DeferredDiagnosticsEmitter`, `RootReason` seems a little confusing to me. Again, just saying. I'm OK if you think the current naming is good enough.

Reply:
I agree that `RootReason` could have a more descriptive name if we can devise one. I don't have particularly good suggestions though.